docs/troubleshooting.md

```diff
@@ -0,0 +1,202 @@
+
+```shell
+$ kubectl logs POD-NAME
+```
+
+Another option can be to review logs stored within the pod. To gain shell access to a pod, run:
+
+```shell
+$ kubectl exec -it POD-NAME -- /bin/bash
+```
+
+> TIP: Turbinia logs can be found in `/mnt/turbiniavolume/logs` and Timesketch logs can be found in `/var/log/timesketch`.
+
+When your pod status is *ImagePullBackOff* or *ErrImagePull*, this means that the pod could not run because it could not pull the image. This typically occurs when there is either an invalid Image or the Kubernetes cluster does not have Internet connectivity. Try pulling the image manually on the node using *docker pull*. For example, to manually pull an image from Docker Hub, use the command below by replacing IMAGE with the image ID:
+
+```shell
+$ docker pull IMAGE
+```
+
+> TIP: When using a private GKE cluster in GCP, please ensure you have a [Cloud NAT](https://cloud.google.com/nat/docs/gke-example#gcloud_3) configured with the network router of your cluster so that third party dependencies can be retrieved.
+
+When your pod status is *Init:CrashLoopBackOff* or *Init:Error*, this means that the pod's init container could not run. To investigate, review the logs of the init container, replacing `POD-NAME` with the pod name and `INIT-POD` with the name of the init pod, then run:
+
+```shell
+$ kubectl logs POD-NAME -c INIT-POD
+```
+
+> TIP: The respective init container name for Turbinia is `init-turbinia` and for Timesketch is `init-timesketch`.
+
+For more troubleshooting tips, see the official k8s docs for [troubleshooting pods](https://kubernetes.io/docs/tasks/debug/debug-application/debug-pods/).
+
+### Troubleshoot Services
+
+Sometimes, when you start a new installation in Kubernetes you may find that a service doesn’t respond when you try to access it, although it was created by a pod in a deployment.
+
+First you can check that the service you are trying to access actually exists, by running the following (replace SVC-NAME with the name of the service you want to access):
+
+```shell
+$ kubectl get SVC-NAME
+```
+
+To get detailed information around the service if it exists, run:
+
+```shell
+$ kubectl describe SVC-NAME
+```
+
+You should see an output message providing some information including any service issues.
+
+If the service is registered and no issues were detected from the step above, it could also be a DNS problem. DNS problems
+typically occurs when a service is in a different namespace then the pod or if the cluster does not have DNS enabled.
+
+To troubleshoot, run the *kubectl get pods* command to get the name of your pod. Then run the command below, this will give you an indication if the DNS resolution is working or not. (remember to replace the `POD-NAME` placeholder with the pod name and `SVC-NAME` placeholder with the name of the service).
+
+```shell
+$ kubectl exec -it POD-NAME nslookup SVC-NAME
```

Comment:
I changed this up to instead refer to the k8s docs as ping wasn't available either in their docs they suggest deploying a pod that has dns utils to help troubleshoot so that should be fine